composables/analytics.js

```diff
@@ -0,0 +1,330 @@
+import dayjs from 'dayjs'
+
+// note: build step can miss unix import for some reason, so
+// we have to import it like this
+// eslint-disable-next-line import/no-named-as-default-member
+const { unix } = dayjs
+
+export const countryCodeToName = (code) => {
+  const regionNames = new Intl.DisplayNames(['en'], { type: 'region' })
+  return regionNames.of(code) || 'Unknown'
+}
+
+export const countryCodeToFlag = (code) => {
+  if (countryCodeToName(code) === 'Unknown') {
```

Comment:
Formatted output is unreliable and should not be used for decision making. According to previous code, fake country code `XX` is used to denote unknown regions, so this check can be more straightforward.

```suggestion
  if (countryCodeToName(code) === 'XX') {
```